proposals/0052-fonts.md

```diff
@@ -0,0 +1,471 @@
+          },
+        ],
+      },
+    ],
+  },
+});
+```
+
+#### Opt-in providers
+
+Other unifont providers are exported from `astro/config`.
+
+```js
+import { defineConfig, fontProviders } from "astro/config";
+
+export default defineConfig({
+  fonts: {
+    providers: [
+      fontProviders.adobe({ apiKey: process.env.ADOBE_FONTS_API_KEY }),
+    ],
+    // ...
+  },
+});
+```
+
+Note that under the hood, the definition would look like:
+
+```ts
+function adobe(config: AdobeConfig): FontProvider {
+  return {
+    name: "adobe",
+    entrypoint: "astro/assets/fonts/adobe",
+    config,
+  };
+}
+
+export const fontProviders = {
+  adobe,
+};
+```
+
+#### Why this API?
+
+1. **Coherent API**: a few things in Astro are using this pattern, namely integrations and vite plugins. It's simple to author as a library author, easy to use as a user
+2. **Keep opt-in providers**: allows to only use 2 providers by default, and keeps the API open to anyone
+3. **Types!**: now that `defineConfig` [supports generics](https://github.com/withastro/astro/pull/12243), we can do powerful things! Associated with type generation, we can generate types for `families` `name`, infer the provider type from `provider` and more.
+
+### Defaults
+
+Astro must provide sensible defaults when it comes to font weights, subsets and more.
+
+We need to decide what default to provide. I can see 2 paths:
+
+| Path      | Example (weight)    | Advantage             | Downside                                                                          |
+| --------- | ------------------- | --------------------- | --------------------------------------------------------------------------------- |
+| Minimal   | Only include `400`  | Lightweight           | People will probably struggle by expecting all weights to be available by default |
+| Extensive | Include all weights | Predictable for users | Heavier by default                                                                |
+
+### Families
+
+A family is made of a least a `name`:
+
+```js
+export default defineConfig({
+  fonts: {
+    families: [
+      {
+        name: "Roboto",
+      },
+      "Roboto", // Shorthand
+    ],
+  },
+});
+```
+
+It can specify options such as `weights` and `subsets`:
+
+```js
+export default defineConfig({
+  fonts: {
+    families: [
+      {
+        name: "Roboto",
+        weights: [400, 600],
+      },
+    ],
+  },
+});
+```
+
+It can also specify a `provider` (and `src` if it's the `local` provider):
+
+```js
+export default defineConfig({
+  fonts: {
+    families: [
+      {
+        name: "Roboto",
+        provider: "local",
+        src: "./Roboto.woff2",
+      },
+    ],
+  },
+});
+```
+
+It also accepts a `as` prop, which will be used when generating CSS. This is useful in the following cases:
+
+1. You are an integration and want to make sure you don't conflict with fonts declared by the user
+2. Your font name can't be properly converted to a CSS variable name
```

Comment:
Who is "you" in these sentences? I think we should rephrase them:

```suggestion
1. Avoid conflicts between the font names declared by integrations and those declared by the end-users.
2. A font name can't be properly converted to a CSS variable name.
```

As for the second bullet point, we should at least provide an example of when a font can't be converted to a CSS variable.

Reply:
I agree with improving the wording! As for point 2, I haven't experienced it but I was thinking about non latin characters (eg. chinese): are they allowed as a css variable name?

Reply:
I found https://www.thingsaboutweb.dev/en/posts/css-variables-part-1 but I'm still unsure. I think there are cases where the variable name contains characters that will be stripped out and `as` will allow the user to have more control

Reply:
We are already using `cssesc` for view transitions, i think we could reuse it to take care of encoding